bin/backup/start-backupjob.sh

```diff
@@ -40,7 +40,11 @@ env_check_info "BACKUP_LABEL" "BACKUP_LABEL is set to ${BACKUP_LABEL}."
 env_check_info "BACKUP_PATH" "BACKUP_PATH is set to ${BACKUP_PATH}."
 env_check_info "BACKUP_OPTS" "BACKUP_OPTS is set to ${BACKUP_OPTS}."
 
-echo "*:*:*:"${BACKUP_USER?}":"${BACKUP_PASS?} >> ${PGPASSFILE?}
+# escape any instances of ':' or '\' with '\' in the provided password
+# before storing the value in the password file
+ESCAPED_PASSWORD=$(sed <<< "${BACKUP_PASS?}" 's/[:\\]/\\&/g')
+
+echo "*:*:*:"${BACKUP_USER?}":"${ESCAPED_PASSWORD?} >> ${PGPASSFILE?}
```

Comment:
I didn't notice that these expansions are outside quotes. As-is, this will mess up values with consecutive whitespace. Instead,

```suggestion
echo "*:*:*:${BACKUP_USER?}:${ESCAPED_PASSWORD?}" >> ${PGPASSFILE?}
```

```
BACKUP_USER="bob"
ESCAPED_PASSWORD="a  b  c"

echo "*:*:*:"${BACKUP_USER?}":"${ESCAPED_PASSWORD?} | hexdump -C
00000000  2a 3a 2a 3a 2a 3a 62 6f  62 3a 61 20 62 20 63 0a  |*:*:*:bob:a b c.|
00000010

echo "*:*:*:${BACKUP_USER?}:${ESCAPED_PASSWORD?}" | hexdump -C
00000000  2a 3a 2a 3a 2a 3a 62 6f  62 3a 61 20 20 62 20 20  |*:*:*:bob:a  b  |
00000010  63 0a                                             |c.|
00000012
```

(The other scripts aren't affected because they use a heredoc.)

Reply:
gotta, good catch, thanks.